test/Math/Rectangle.test.cpp

```diff
@@ -61,6 +61,34 @@ TEST(Rectangle, translate) {
 	EXPECT_EQ((NAS2D::Rectangle{4, 6, 5, 6}), (NAS2D::Rectangle{3, 4, 5, 6}).translate({1, 2}));
 }
 
+TEST(Rectangle, operatorAddSubtract) {
+	//translate(...) and operator overloads are equivalent.
+	EXPECT_EQ((NAS2D::Rectangle{0, 0, 1, 1}).translate({1, 1}), (NAS2D::Rectangle{0, 0, 1, 1} + Vector<int>{1, 1}));
+	EXPECT_EQ((NAS2D::Rectangle{1, 1, 1, 1}).translate({1, 1}), (NAS2D::Rectangle{1, 1, 1, 1} + Vector<int>{1, 1}));
+	EXPECT_EQ((NAS2D::Rectangle{3, 4, 5, 6}).translate({1, 2}), (NAS2D::Rectangle{3, 4, 5, 6} + Vector<int>{1, 2}));
+
+	//Translate by a negative vector is the same as adding a negative or subtracting a positive.
+	EXPECT_EQ((NAS2D::Rectangle{1, 1, 1, 1}).translate({-1, -1}), (NAS2D::Rectangle{1, 1, 1, 1} + Vector<int>{-1, -1}));
+	EXPECT_EQ((NAS2D::Rectangle{1, 1, 1, 1}).translate({-1, -1}), (NAS2D::Rectangle{1, 1, 1, 1} - Vector<int>{1, 1}));
+	EXPECT_EQ((NAS2D::Rectangle{3, 4, 5, 6}).translate({-1, 2}), (NAS2D::Rectangle{3, 4, 5, 6} + Vector<int>{1, 2}));
+
+	//Compound operators are the same as translating a rectangle then assigning it to itself...for now
+	{
+		auto r1 = NAS2D::Rectangle{1, 1, 1, 1};
+		r1 = r1.translate({1, 1});
+		auto r2 = NAS2D::Rectangle{1, 1, 1, 1};
+		r2 += Vector<int>{1, 1};
+		EXPECT_EQ(r1, r2);
```

Comment:
I would suggest doing the assignment to `r1` as a single step here. I suppose maybe you're illustrating the similarity between the two operations here, though I don't feel the intermediate steps really adds much. Mostly it leaves me wondering why it was written that way, and if I need to watch out for something.

I suppose part of my concern is the unit tests are sometimes used as example code to show how a feature is used, and the assignment to `r1` is rather odd here.